### PR TITLE
Fix converging

### PR DESCRIPTION
**Pull request by doublej472**
This fixes converging on test kitchen by adding `https://supermarket.chef.io` as a source and constraining the mysql and databaase cookbooks.

### CODE REVIEW

File: Berksfile [hunk 1]
```diff
@@ -1,4 +1,5 @@
 source 'http://api.berkshelf.com'
+source 'https://supermarket.chef.io'
```
Comment (ramereth) [inline Berksfile]: Remove the line above as well as this one replaces that.

File: metadata.rb [hunk 2]
```diff
@@ -11,11 +11,11 @@
 depends          'python'
 depends          'git'
 depends          'nginx'
-depends          'mysql'
+depends          'mysql', '= 3.0.12'
```
Comment (ramereth) [inline metadata.rb]: I wouldn't lock exact versions but instead use ``~>`` instead.

File: metadata.rb [hunk 3]
```diff
@@ -11,11 +11,11 @@
...
 depends          'openssl'
 depends          'postgresql'
 depends          'sqlite'
-depends          'database'
+depends          'database', '= 1.4.0'
```
Comment (ramereth) [inline metadata.rb]: Same here with using ``~>`` instead.

Comment (jldugger) [inline metadata.rb]: might as well use pessimistic depends (`~>`) for this PR.

**Comment by doublej472** [comment]
@jldugger @ramereth Now this is using pessimistic depends as requested. Also, I found that the berkshelf source works if you change the protocol to `https`, so adding the supermarket repo is unnecessary.

**Comment by fahlmant** [comment]
@doublej472 Our standard is to use the supermarket URL now anyway, so please use that.

**Comment by jldugger** [comment]
@doublej472 I think we want to use supermarket, not berkshelf.com.

**Comment by doublej472** [comment]
@jldugger @fahlmant This is now using `https://supermarket.chef.io` as a source instead of `https://api.berkshelf.com`

**Comment by ramereth** [comment]
@doublej472 can you go ahead and add the following line to metadata.rb so that it only tests CentOS 6?

``` ruby
supports         'centos', '~> 6.0'
```

Also, can you just remove the default TK suite and rename the mysql one to default?

**Comment by ramereth** [comment]
Mind update the ServerSpec so that it uses the newer of invocation?

**Comment by doublej472** [comment]
@ramereth what do you mean? The 2 tests that exist weren't used before.

File: .kitchen.yml [hunk 4]
```diff
@@ -17,36 +17,12 @@ platforms:
...
       box_url: http://vagrant.osuosl.org/centos-6-20140121.box
 
-  - name: ubuntu-12.04
-    driver_plugin: vagrant
-    driver_config:
-      box: ubuntu-12.04-20140121
-      box_url: http://vagrant.osuosl.org/ubuntu-12.04-20140121.box
-    run_list:
-      - recipe[apt::default]
-
```
Comment (ramereth) [inline .kitchen.yml]: You might as well remove all vagrant references too.

**Comment by ramereth** [comment]
@doublej472 The existing tests should still be valid for what was the mysql suite. But you need to replace the ``include`` lines with one ``set :backend, :exec`` so that when you run kitchen verify it runs properly.

**Comment by doublej472** [comment]
The tests work now, but it certainly doesn't pass them, specifically the ports aren't listening for connections.

**Comment by ramereth** [comment]
@doublej472 @Kennric just merged osuosl/ganeti_webmgr#103 so this should work now. You might verify yourself to be sure.

**Comment by jldugger** [comment]
When I `kitchen converge` this, it fails on vncauthproxy timeout.

<details>

```             
             ================================================================================
             Error executing action `run` on resource 'ruby_block[restart_service]'
             ================================================================================
             
             Mixlib::ShellOut::ShellCommandFailed
             ------------------------------------
             Expected process to exit with [0], but received '1'
             ---- Begin output of /sbin/sv restart /etc/service/vncauthproxy ----
             STDOUT: timeout: down: /etc/service/vncauthproxy: 0s, normally up, want up
             STDERR: 
             ---- End output of /sbin/sv restart /etc/service/vncauthproxy ----
             Ran /sbin/sv restart /etc/service/vncauthproxy returned 1
             
             Cookbook Trace:
             ---------------
             /tmp/kitchen/cookbooks/runit/libraries/helpers.rb:174:in `tap'
             /tmp/kitchen/cookbooks/runit/libraries/helpers.rb:174:in `safe_sv_shellout!'
             /tmp/kitchen/cookbooks/runit/libraries/helpers.rb:202:in `restart_service'
             /tmp/kitchen/cookbooks/runit/libraries/provider_runit_service.rb:56:in `block (3 levels) in <class:RunitService>'
             /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:78:in `run_action'
             /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:136:in `run_delayed_notification'
             /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:124:in `block in run_delayed_notifications'
             /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:123:in `each'
             /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:123:in `run_delayed_notifications'
             /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:113:in `converge'
             /tmp/kitchen/cookbooks/runit/libraries/provider_runit_service.rb:282:in `block in <class:RunitService>'
             /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:78:in `run_action'
             /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block (2 levels) in converge'
             /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `each'
             /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block in converge'
             /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:105:in `converge'
             
             Resource Declaration:
             ---------------------
             # In /tmp/kitchen/cookbooks/runit/libraries/provider_runit_service.rb
             
       53:         ruby_block 'restart_service' do
       54:           block do
       55:             action_enable
       56:             restart_service
       57:           end
       58:           action :nothing
       59:           only_if { new_resource.restart_on_update && !new_resource.start_down }
       60:         end
       61: 
             
             Compiled Resource:
             ------------------
             # Declared in /tmp/kitchen/cookbooks/runit/libraries/provider_runit_service.rb:53:in `block in <class:RunitService>'
             
             ruby_block("restart_service") do
        action [:nothing]
        retries 0
        retry_delay 2
        default_guard_interpreter :default
        block_name "restart_service"
        declared_type :ruby_block
        cookbook_name :ganeti_webmgr
        block #<Proc:0x000000081bd0c0@/tmp/kitchen/cookbooks/runit/libraries/provider_runit_service.rb:54>
        only_if { #code block }
             end
             
             Platform:
             ---------
             x86_64-linux
             
           * ruby_block[restart_log_service] action run
             * ruby_block[restart_service] action nothing (skipped due to action :nothing)
             * ruby_block[restart_log_service] action nothing (skipped due to action :nothing)
             * directory[/etc/sv/vncauthproxy] action create (up to date)
             * template[/etc/sv/vncauthproxy/run] action create (up to date)
             * directory[/etc/sv/vncauthproxy/log] action create (up to date)
             * directory[/etc/sv/vncauthproxy/log/main] action create (up to date)
             * directory[/var/log/vncauthproxy] action create (up to date)
             * template[/etc/sv/vncauthproxy/log/config] action create (up to date)
             * link[/var/log/vncauthproxy/config] action create (up to date)
             * template[/etc/sv/vncauthproxy/log/run] action create (up to date)
             * directory[/etc/sv/vncauthproxy/env] action create (up to date)
             * ruby_block[Delete unmanaged env files for vncauthproxy service] action run (skipped due to only_if)
             * template[/etc/sv/vncauthproxy/check] action create (skipped due to only_if)
             * template[/etc/sv/vncauthproxy/finish] action create (skipped due to only_if)
             * directory[/etc/sv/vncauthproxy/control] action create (up to date)
             * link[/etc/init.d/vncauthproxy] action create (up to date)
             * file[/etc/sv/vncauthproxy/down] action nothing (skipped due to action :nothing)
             * directory[/etc/service] action create (up to date)
             * link[/etc/service/vncauthproxy] action create (up to date)
             * ruby_block[wait for vncauthproxy service socket] action run
        - execute the ruby block wait for vncauthproxy service socket
             - execute the ruby block restart_log_service
           
           ================================================================================
           Error executing action `enable` on resource 'runit_service[vncauthproxy]'
           ================================================================================
           
           Mixlib::ShellOut::ShellCommandFailed
           ------------------------------------
           ruby_block[restart_service] (/tmp/kitchen/cookbooks/runit/libraries/provider_runit_service.rb line 53) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
           ---- Begin output of /sbin/sv restart /etc/service/vncauthproxy ----
           STDOUT: timeout: down: /etc/service/vncauthproxy: 0s, normally up, want up
           STDERR: 
           ---- End output of /sbin/sv restart /etc/service/vncauthproxy ----
           Ran /sbin/sv restart /etc/service/vncauthproxy returned 1
           
           Cookbook Trace:
           ---------------
           /tmp/kitchen/cookbooks/runit/libraries/helpers.rb:174:in `tap'
           /tmp/kitchen/cookbooks/runit/libraries/helpers.rb:174:in `safe_sv_shellout!'
           /tmp/kitchen/cookbooks/runit/libraries/helpers.rb:202:in `restart_service'
           /tmp/kitchen/cookbooks/runit/libraries/provider_runit_service.rb:56:in `block (3 levels) in <class:RunitService>'
           /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:78:in `run_action'
           /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:136:in `run_delayed_notification'
           /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:124:in `block in run_delayed_notifications'
           /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:123:in `each'
           /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:123:in `run_delayed_notifications'
           /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:113:in `converge'
           /tmp/kitchen/cookbooks/runit/libraries/provider_runit_service.rb:282:in `block in <class:RunitService>'
           /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:78:in `run_action'
           /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block (2 levels) in converge'
           /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `each'
           /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block in converge'
           /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:105:in `converge'
           
           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cookbooks/ganeti_webmgr/recipes/default.rb
           
           140: runit_service 'vncauthproxy' do
           141:   options(
           142:     'port' => node['ganeti_webmgr']['vncauthproxy']['port'],
           143:     'ip' => node['ganeti_webmgr']['vncauthproxy']['ip'],
           144:     'install_dir' => node['ganeti_webmgr']['install_dir']
           145:   )
           146: end
           147: 
           
           Compiled Resource:
           ------------------
           # Declared in /tmp/kitchen/cookbooks/ganeti_webmgr/recipes/default.rb:140:in `from_file'
           
           runit_service("vncauthproxy") do
             provider Chef::Provider::RunitService
             action :enable
             updated true
             updated_by_last_action true
             supports {:restart=>true, :reload=>true, :status=>true}
             retries 0
             retry_delay 2
             default_guard_interpreter :default
             service_name "vncauthproxy"
             pattern "vncauthproxy"
             status_command "/sbin/sv status /etc/service"
             sv_bin "/sbin/sv"
             sv_dir "/etc/sv"
             service_dir "/etc/service"
             lsb_init_dir "/etc/init.d"
             options {"port"=>"8888", "ip"=>"0.0.0.0", "install_dir"=>"/opt/ganeti_webmgr"}
             log true
             restart_on_update true
             run_template_name "vncauthproxy"
             log_template_name "vncauthproxy"
             check_script_template_name "vncauthproxy"
             finish_script_template_name "vncauthproxy"
             sv_templates true
             log_dir "/var/log/vncauthproxy"
             service_mirror # Declared in 
           
           service("vncauthproxy") do
             provider Chef::Provider::Service::Simple
             action [:nothing]
             supports {:restart=>true, :reload=>true, :status=>true}
             retries 0
             retry_delay 2
             default_guard_interpreter :default
             service_name "vncauthproxy"
             pattern "vncauthproxy"
             start_command "/sbin/sv start /etc/service/vncauthproxy"
             stop_command "/sbin/sv stop /etc/service/vncauthproxy"
             status_command "/sbin/sv status /etc/service/vncauthproxy"
             restart_command "/sbin/sv restart /etc/service/vncauthproxy"
           end
           
             declared_type :runit_service
             cookbook_name :ganeti_webmgr
             recipe_name "default"
           end
           
           Platform:
           ---------
           x86_64-linux
           
       Recipe: mysql::server
         * service[mysql] action restart
           - restart service service[mysql]

```
</details>

When I run it a second time, it converges, presumably because the `only_if` doesn't trigger a restart (when it arguably should). Is this something we want to try to fix now?

**Comment by ramereth** [comment]
@jldugger doh! I forgot that the cookbook defaults to the develop branch. My fix went into master. I just created osuosl/ganeti_webmgr#104 to resolve this.

**Comment by doublej472** [comment]
This verifies successfully now, but Twisted seems to take long enough to time-out my ssh session during converge. I can still login and see chef-client running, and when I re-converge it again it succeeds. Is there a way to configure ssh to not drop the connection on long kitchen runs?

**Comment by ramereth** [comment]
@doublej472 that's related to the ssh timeout bug we keep talking about. When that happens again, just log into the vm, kill chef-client and then logout and run converge again.

**Comment by doublej472** [comment]
@ramereth should I wait for another review on this since you dismissed the review done by @jldugger ?

**Comment by ramereth** [comment]
@doublej472 wait until osuosl/chef-repo#795 has been merged before you merge this. And then only bump the gwm environment. i.e. ``!bump minor gwm``.

**Comment by doublej472** [comment]
!bump minor gwm

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v0.5.0.  Have a nice day!

**Pull request merged**